File: Atcoder-archive/ABC_083/d.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
#include <complex>

using namespace std;
typedef long long ll;

int main(){
  string s;
  cin >> s;
  char b = s[0];
  ll maxL = s.size();
  for(auto i=1;i<s.size();i++){
    //違う部分を探す。
    if(s[i] != b){
      auto l = max(i, int(s.size() - i) );
      maxL = min(maxL,l)
      b = s[i]
    }
  }
}

File: Atcoder-archive/ABC_083/d.py
S = input()

# s = S[0]
temp_T = len(S)
for i in range(len(S)-1):
    if(S[i] != S[i+1]):
        judge = max(i+1,len(S)-(i+1))
        temp_T = min(judge, temp_T)

print(temp_T)¥
